File: src/jastudio_src/jastudio/viewmodels/kanji_list/kanji_list_viewmodel.py
from __future__ import annotations

from typing import TYPE_CHECKING, override

from autoslot import Slots  # pyright: ignore[reportMissingTypeStubs]

if TYPE_CHECKING:
    from jastudio.viewmodels.kanji_list.sentence_kanji_viewmodel import KanjiViewModel


class KanjiListViewModel(Slots):
    def __init__(self, kanji_list: list[KanjiViewModel]) -> None:
        self.kanji_list: list[KanjiViewModel] = kanji_list

    @override
    def __str__(self) -> str: return "\n".join([str(kan) for kan in self.kanji_list])

File: src/src_dotnet/JAStudio.Core/ViewModels/KanjiList/KanjiListViewModel.cs
using System.Collections.Generic;
using System.Linq;

namespace JAStudio.Core.ViewModels.KanjiList;

public class KanjiListViewModel
{
   public List<KanjiViewModel> KanjiList { get; }

   public KanjiListViewModel(List<KanjiViewModel> kanjiList) => KanjiList = kanjiList;

   public override string ToString()
   {
      return string.Join("\n", KanjiList.Select(kan => kan.ToString()));
   }
}
